packages/user/Tokens/query-service/src/lib.rs

```diff
@@ -44,6 +45,22 @@ mod service {
             }
             Ok(())
         }
+
+        fn get_subaccount(
+            &self,
+            user: AccountNumber,
+            sub_account: &str,
+        ) -> async_graphql::Result<SubAccount> {
+            SubAccountTable::with_service(tokens::SERVICE)
+                .get_index_pk()
+                .get(&(user, sub_account.to_string()))
+                .ok_or_else(|| {
+                    async_graphql::Error::new(format!(
+                        "Sub-account '{}' not found for user '{}'",
+                        sub_account, user
+                    ))
```

Comment:
I kinda prefer just a straight `Option`. 
Otherwise this can tank the rest of the Graphql query which is annoying.